### ISSUE TITLE
bad output of CAST AS JSON for nested array of decimals.

### ISSUE THREAD

**Issue by losipiuk** [prestodb/presto#7328]
If we cast array of array of decimals to JSON we get the inner value wrapped in extra quotes.

**How to reproduce:**

```
select cast (array[array[decimal '1234567890.1234567890234567890']] as JSON);
                _col0
--------------------------------------
 [["1234567890.1234567890234567890"]]
```

**Expected behavior**
```
select cast (array[array[decimal '1234567890.1234567890234567890']] as JSON);
                _col0
--------------------------------------
 [[1234567890.1234567890234567890]]
```

**Comment by dain** [comment]
That is an interesting problem.  If you think of JSON as an extension of JavaScript, then a number is a 32bit float, so adding quotes makes the client preserve the full data.  If you think if JSON as a completely independent data format, then the quotes make it a string.  IIRC @martint dealt with this before.

**Comment by cawallin** [comment]
Fixed by #7565 ( @losipiuk, feel free to re-open if that's not the case)
